core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -312,28 +312,31 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * }}}
     */
   def bufferBy(f: O => Boolean): Stream[F, O] = {
-    def dumpBuffer(bb: List[Chunk[O]]): Pull[F, O, Unit] =
-      bb.reverse.foldLeft(Pull.done: Pull[F, O, Unit])((acc, c) => acc >> Pull.output(c))
+    @inline
+    def dumpBuffer(bb: ListBuffer[Chunk[O]], rest: => Pull[F, O, Unit]): Pull[F, O, Unit] =
+      if (bb.isEmpty) rest else Pull.output(bb.head) >> dumpBuffer(bb.tail, rest)
 
-    def go(buffer: List[Chunk[O]], last: Boolean, s: Stream[F, O]): Pull[F, O, Unit] =
+    def go(buffer: ListBuffer[Chunk[O]], last: Boolean, s: Stream[F, O]): Pull[F, O, Unit] =
```

Comment:
AFAIK, half a year ago Chunk was redesigned to achieve effective splitAt & merge. Shouldn't we try to write this method with Chunk & Chunk.Queue, but without Buffers & Vectors?

Reply:
@diesalbla Any thoughts on updating this to use Chunk or Chunk.Queue? I vaguely remember trying once before and not liking the result. I'm curious about the use of ListBuffer though.